libs/framework/src/celix_bundle_cache.c

```diff
@@ -71,124 +137,223 @@ celix_status_t celix_bundleCache_create(celix_framework_t* fw, celix_bundle_cach
+    }
+    celixThreadMutex_unlock(&fw->cache->mutex);
+    if (bndId == -1) {
+        celix_bundleCache_updateIdForLocationLookupMap(fw);
+        celixThreadMutex_lock(&fw->cache->mutex);
+        bndId = celix_stringHashMap_getLong(fw->cache->locationToBundleIdLookupMap, location, -1);
+        celixThreadMutex_unlock(&fw->cache->mutex);
+    }
+    return bndId;
+}
 
-	framework_logIfError(celix_frameworkLogger_globalLogger(), status, NULL, "Failed to get bundle archives");
-	if (status != CELIX_SUCCESS) {
-		perror("\t");
-	}
+bool celix_bundleCache_isBundleIdAlreadyUsed(celix_framework_t *fw, long bndId) {
+    bool found = false;
+    celixThreadMutex_lock(&fw->cache->mutex);
+    CELIX_STRING_HASH_MAP_ITERATE(fw->cache->locationToBundleIdLookupMap, iter) {
+        if (iter.value.longValue == bndId) {
+            found = true;
+            break;
+        }
+    }
+    celixThreadMutex_unlock(&fw->cache->mutex);
+    return found;
+}
 
-	return status;
+
+static celix_status_t celix_bundleCache_createBundleArchivesForSpaceSeparatedList(celix_framework_t *fw, long* bndId, const char* list, bool logProgress) {
+    celix_status_t status = CELIX_SUCCESS;
+    char delims[] = " ";
+    char *savePtr = NULL;
+    char zipFileListBuffer[CELIX_DEFAULT_STRING_CREATE_BUFFER_SIZE];
+    char* zipFileList = celix_utils_writeOrCreateString(zipFileListBuffer, sizeof(zipFileListBuffer), "%s", list);
+    if (zipFileList) {
+        char *location = strtok_r(zipFileList, delims, &savePtr);
+        while (location != NULL) {
+            bundle_archive_t* archive = NULL;
+            status = celix_bundleCache_createArchive(fw, (*bndId)++, location, &archive);
+            if (status != CELIX_SUCCESS) {
+                fw_logCode(fw->logger, CELIX_LOG_LEVEL_ERROR, CELIX_BUNDLE_EXCEPTION, "Cannot create bundle archive for %s", location);
+                break;
+            } else {
+                celix_log_level_e lvl = logProgress ? CELIX_LOG_LEVEL_INFO : CELIX_LOG_LEVEL_DEBUG;
+                fw_log(fw->logger, lvl, "Created bundle cache '%s' for bundle archive %s (bndId=%li).",
+                       celix_bundleArchive_getCurrentRevisionRoot(archive),
+                       celix_bundleArchive_getSymbolicName(archive), celix_bundleArchive_getId(archive));
+                bundleArchive_destroy(archive);
+            }
+            location = strtok_r(NULL, delims, &savePtr);
+        }
+    } else {
+        status = CELIX_ENOMEM;
+        fw_logCode(fw->logger, CELIX_LOG_LEVEL_ERROR, status, "Failed to create zip file list.");
+    }
+    celix_utils_freeStringIfNeeded(zipFileListBuffer, zipFileList);
+    return status;
 }
 
-celix_status_t celix_bundleCache_createArchive(celix_framework_t* fw, long id, const char * location, const char *inputFile, bundle_archive_pt *archive) {
-	celix_status_t status = CELIX_SUCCESS;
-	char archiveRoot[512];
+//using tmp cache and that bundle cache is not deleted during bundle cache destroy.
```

Comment:
```suggestion
//Not using tmp cache and that bundle cache is not deleted during bundle cache destroy.
```

Is that what you mean?